pkg/operator/controllers/workaround/systemreserved.go

```diff
@@ -84,29 +85,52 @@ func (sr *systemreserved) kubeletConfig() (*mcv1.KubeletConfig, error) {
 func (sr *systemreserved) Ensure(ctx context.Context) error {
 	// Step 1. Add label to worker MachineConfigPool.
 	// Get the worker MachineConfigPool, modify it to add a label aro.openshift.io/limits: "", and apply the modified config.
-	mcp, err := sr.mcocli.MachineconfigurationV1().MachineConfigPools().Get(ctx, workerMachineConfigPoolName, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	// don't update if we don't need to.
-	if _, ok := mcp.Labels[labelName]; !ok {
-		if mcp.Labels == nil {
-			mcp.Labels = map[string]string{}
-		}
-		mcp.Labels[labelName] = labelValue
-
-		_, err = sr.mcocli.MachineconfigurationV1().MachineConfigPools().Update(ctx, mcp, metav1.UpdateOptions{})
+	var changed bool
+	for _, mcpName := range []string{masterMachineConfigPoolName, workerMachineConfigPoolName} {
+		mcp, err := sr.mcocli.MachineconfigurationV1().MachineConfigPools().Get(ctx, mcpName, metav1.GetOptions{})
 		if err != nil {
+			// if machineSet does not exist - skip
+			if kerrors.IsNotFound(err) {
+				continue
+			}
 			return err
 		}
+		// don't update if we don't need to.
+		if _, ok := mcp.Labels[labelName]; !ok {
+			if mcp.Labels == nil {
+				mcp.Labels = map[string]string{}
+			}
+			mcp.Labels[labelName] = labelValue
+
+			_, err = sr.mcocli.MachineconfigurationV1().MachineConfigPools().Update(ctx, mcp, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+			changed = true
+		}
+	}
+
+	// If there is no KubeletConfig - mark it as changed too
+	_, err := sr.mcocli.MachineconfigurationV1().KubeletConfigs().Get(ctx, kubeletConfigName, metav1.GetOptions{})
+	if kerrors.IsNotFound(err) {
+		changed = true
 	}
 
-	//   Step 2. Create KubeletConfig CRD with appropriate limits.
-	kc, err := sr.kubeletConfig()
+	//  Step 2. Create KubeletConfig CRD with appropriate limits.
+	kc, err := kubeletConfig()
 	if err != nil {
 		return err
 	}
 
+	// we have to recreate due to https://bugzilla.redhat.com/show_bug.cgi?id=1995621
+	_, err = sr.mcocli.MachineconfigurationV1().KubeletConfigs().Get(ctx, kubeletConfigName, metav1.GetOptions{})
+	if err == nil && changed { // exits and change detected - recreate
+		err := sr.mcocli.MachineconfigurationV1().KubeletConfigs().Delete(ctx, kubeletConfigName, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
 	return sr.dh.Ensure(ctx, kc)
```

Comment:
Im not a fan of keeping this when same API clients got introduced. But few reasons why:
1. API client shenanigans above are just because of  https://bugzilla.redhat.com/show_bug.cgi?id=1995621 and should go away eventually.
2.  Ensure handle object updates in a nice way with merge. If we ever update `KubeletConfig` with new values this will make sure it is updated. If moved to client calls we would need to handle this ouselfs, ending up with few more api calls.

Reply:
I agree with this statement. Let us keep it here. Would you think there would be any value in putting the `// TODO: Remove once https://bugzilla.redhat.com/show_bug.cgi?id=1995621 gone`? I am afraid that these types of fixes are easily forgettable.